File: Euclid/Messages/Outgoing/FLATCREATED.cs
using Euclid.Network;
using Euclid.Network.Streams.Util;
using Euclid.Storage.Database.Data;

namespace Euclid.Messages.Outgoing
{
    class FLATCREATED : IMessageComposer
    {
        private RoomData roomData;

        public FLATCREATED(RoomData roomData)
        {
            this.roomData = roomData;
        }

        public override void Write()
        {
            Data.Add(new ArgumentEntry(roomData.Id));
            Data.Add(new DelimeterEntry(GameServer.Instance.PrivateServer.IpAddress, " "));
            Data.Add(new DelimeterEntry(GameServer.Instance.PrivateServer.Port, " "));
            Data.Add(new DelimeterEntry(roomData.Name, " "));
        }

    }
}

File: euclid/src/main/java/euclid/messages/outgoing/FLATCREATED.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.GameServer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;
import euclid.storage.database.data.RoomData;

public class FLATCREATED extends IMessageComposer {

    private final RoomData roomData;

    public FLATCREATED(RoomData roomData) {
        this.roomData = roomData;
    }

    @Override
    public void write() {
        GameServer server = GameServer.getInstance();
        getData().add(new ArgumentEntry(roomData.getId()));
        getData().add(new DelimeterEntry(server.getPrivateServer().getIpAddress(), " "));
        getData().add(new DelimeterEntry(server.getPrivateServer().getPort(), " "));
        getData().add(new DelimeterEntry(roomData.getName(), " "));
    }
}
